### PR TITLE
LGA-820 - Switch to Redis

### PR DESCRIPTION
**Pull request by sldblog**
## What does this pull request do?

Reconfigures the application to use Redis as the Celery broker.

Cloud Platform staging namespace has been provisioned with Amazon ElastiCache for Redis to match this change:

- https://github.com/ministryofjustice/cloud-platform-environments/pull/1182
- https://github.com/ministryofjustice/cloud-platform-environments/pull/1184

## Any other changes that would benefit highlighting?

Please review commit-by-commit to review the intent in the commit messages.

Performance with node_type cache.t2.micro
redis-cp: 10 minutes with 8 workers for 4700 records, 470/minute
amqp-td: 2.5 minutes with 8 workers for 4700 records, 1880/minute

redis-cp is 4x slower than amqp-td but at this stage we are trying to get the application to work.
Further work is required to address performance when needed

## Checklist

- [x] Provided JIRA ticket number in the title, e.g. "LGA-152: Sample title".
- [x] 🚨In this form, this change is **incompatible** with template-deploy. Template-deploy needs to be reconfigured to use `CELERY_BROKER_URL` for RabbitMQ (and installing the necessary libraries) for it to work.
- [ ] ⚠️The production Cloud Platform namespace needs an ElastiCache instance.

### CODE REVIEW

File: laalaa/settings/base.py [hunk 1]
```diff
@@ -156,6 +157,10 @@ def root(*x):
 
 CELERY_BROKER_URL = os.environ.get("CELERY_BROKER_URL", "redis://localhost:6379")
 
+CELERY_BROKER_USE_SSL = os.environ.get("CELERY_BROKER_USE_SSL", None)
+if CELERY_BROKER_USE_SSL:
+    CELERY_BROKER_USE_SSL = {"ssl_cert_reqs": ssl.CERT_NONE}
```
Comment (said-moj) [inline laalaa/settings/base.py]: [feedback from David] We might need to review certificate management

File: kubernetes_deploy/staging/queue_worker_deployment.yml [hunk 2]
```diff
@@ -5,7 +5,7 @@ metadata:
     kubernetes.io/change-cause: "<to be filled in deploy_to_kubernetes script>"
   name: laa-legal-adviser-api-worker
 spec:
-  replicas: 1
+  replicas: 4
```
Comment (sldblog) [inline kubernetes_deploy/staging/queue_worker_deployment.yml]: Could you explain in the commit on why this change was made, please? During standup, you mentioned that in-pod concurrency was not making much of a difference, which would be valuable context here.

What is the time to upload how many records with one and with four replicas? It would be great to establish our desired target time to load a specific test spreadsheet.

Comment (sldblog) [reply kubernetes_deploy/staging/queue_worker_deployment.yml]: Additionally, we default the `WORKER_APP_CONCURRENCY` setting to 2, which may be way too low. We could increase that to 4, 8, 16, etc., balancing between resource use and pods.

Comment (said-moj) [reply kubernetes_deploy/staging/queue_worker_deployment.yml]: I Reset the `replicas` to 1 and increased WORKER_APP_CONCURRENCY to 8 to avoid over provisioning. Hopefully when production starts using node_type of `cache.t2.medium` the performance figures in the PR body will improve.

Comment (sldblog) [reply kubernetes_deploy/staging/queue_worker_deployment.yml]: I'm confused, it still says 4 in cf928ce

**Comment by sldblog** [comment]
:tophat: :bowing_man:
